File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let output = 0
    let buyPrice = prices[0]
    for (let i = 1; i < prices.length; i++) {
        const price = prices[i]
        if (price < buyPrice) {
            buyPrice = price
        } else if (price > buyPrice) {
            let temp = price - buyPrice
            if (temp > output) {
                output = temp
            }
        }
    }

    return output
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.go
func maxProfit(prices []int) int {
    output := 0
    buyPrice := prices[0]
    for i:= 1; i < len(prices); i++ {
        price := prices[i]
        if price < buyPrice {
            buyPrice = price
        } else if price > buyPrice {
            temp := price - buyPrice
            if temp > output {
                output = temp
            }
        }
    }

    return output
}